content/blog/how-to-ask-good-questions/index.md

```diff
@@ -0,0 +1,110 @@
+---
+{
+	title: "How to ask better questions",
+	description: "We all ask questions from time to time, so here are some of my favourite tips when it comes to how to improve the quality of your questions.",
+	published: '2022-07-20T20:10:03.000Z',
+	authors: ['alexchadwick'],
+	tags: ['opinion'],
+	attached: [],
+	license: 'cc-by-nc-sa-4'
+}
+---
+
+
+## Introduction
+
+In this day and age, the programming ecosystem has become so rich and complex that asking questions is inevitable for any developer, regardless of what stage of their career they find themselves in.
+
+However, it is not always easy to ask a question, and often you'll see StackOverflow questions being downvoted or taken down due to the question being of poor quality.
+
+Asking the best possible questions is essential to getting help quickly, as whoever is reading your question will have immediate access to most of the information they need to come up with an answer that could potentially answer your doubts.
+
+So in this article, I wanted to cover the basics on how to improve your questions, in order to get good answers.
+
+Note: Most of these concepts could also apply to opening issues on OSS projects, or to give feedback to different pieces of software
+
+## Define the exact problem you're facing
+
+Now whilst this sounds simple, there are many people who struggle to define their problem. If you're not able to identify the problem you've got, how do you expect other people to be able to identify it for you?
+
+The most basic way to define your problem is to explain what the expected behaviour is, and what the actual behaviour is. For example:
+
+> *I've written a function `sum` that is meant to return the sum of 2 numbers. When I do `sum(1, 1);` I'm expecting it to return 2, however it's returning 1.*
+
+Now whoever is reading my question knows what I'm trying to achieve and what is currently happening.
+
+I see a lot of questions where people describe their expected behaviour but omit their current behaviour, or the other way around.
+
+Also feel free to share screenshots if whatever you're working on has graphics, however **never** use screenshots for code, output, or errors that may display on the command line. Most people prefer that you copy and paste them instead, and it ensures it remains accessible for people with disabilities and translatable for people that speak different languages. Only use images when you need to if possible.
```

Comment:
```suggestion
Also, feel free to share screenshots if whatever you're working on has graphics, however, **always try to avoid using screenshots for code, output, or errors that may display on the command line**. Most people prefer that you copy and paste them instead, as it ensures it remains accessible for people with disabilities and translatable for people that speak different languages. Only use images when you need to if possible.
```